### PR TITLE
DEX-290: customFields patching/testing

### PR DESCRIPTION
**Pull request by naknomum**
## Pull Request Overview

- handles PATCH on Sightings for path=/customFields

---

**Review Notes**
- adds some test util for modifying EDM configuration in order to create a customField to test with

### CODE REVIEW

File: app/modules/sightings/resources.py [hunk 1]
```diff
@@ -476,20 +480,26 @@ def patch(self, args, sighting):
...
+        # no EDM, so fall thru to regular houston-patching
+        context = api.commit_or_abort(
+            db.session, default_error_message='Failed to update Sighting details.'
+        )
+        with context:
+            parameters.PatchSightingDetailsParameters.perform_patch(args, obj=sighting)
+            db.session.merge(sighting)
+        # this mimics output format of edm-patching
+        return {
+            'success': True,
```
Comment (hwindsor) [inline app/modules/sightings/resources.py:500]: What is the purpose of this success field? I've seen it elsewhere too. It seems just to be a less detailed duplication of the status code. EDM may have had this but I don't understand why houston should.

File: app/modules/sightings/resources.py [hunk 2]
```diff
@@ -476,20 +480,26 @@ def patch(self, args, sighting):
...
                     sighting.version = new_version
-                with db.session.begin():
-                    db.session.merge(sighting)
+                    context = api.commit_or_abort(
+                        db.session,
+                        default_error_message='Failed to update Sighting version.',
+                    )
+                    with context:
+                        db.session.merge(sighting)
+            return rdata
```
Comment (hwindsor) [inline app/modules/sightings/resources.py:489]: Re other comment, why not just return rdata['result']?

File: deploy/codex/README.md [hunk 3]
```diff
@@ -45,7 +45,7 @@ Precision nuke example:
 
     docker-compose stop houston && docker-compose rm -f houston && docker volume rm codex_houston-var
 
-Docker is conservative about cleaning up unused objects. This can cause Docker to run out of disk space or 
+Docker is conservative about cleaning up unused objects. This can cause Docker to run out of disk space or
```
Comment (hwindsor) [inline deploy/codex/README.md:48]: Is this file intentional or a rebase error?

File: tests/extensions/edm/utils.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+"""
+EDM utils
+-------------
+"""
+import json
+
+# from tests import utils as test_utils
```
Comment (hwindsor) [inline tests/extensions/edm/utils.py:8]: If they're not used, remove :)

File: tests/extensions/edm/utils.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    expected_status_code=200,
+):
+    with flask_app_client.login(user, auth_scopes=('configuration:write',)):
+        response = flask_app_client.post(
+            '%s' % CONFIG_PATH,
+            data=json.dumps(data_in),
+            content_type='application/json',
+        )
+    assert response.status_code == expected_status_code
+    assert isinstance(response.json, dict)
```
Comment (hwindsor) [inline tests/extensions/edm/utils.py:30]: any reason you're not using the test_utils.validate_dict_response?

File: tests/extensions/edm/utils.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+def configuration_post(
+    flask_app_client,
+    user,
+    data_in={},
+    expected_status_code=200,
+):
+    with flask_app_client.login(user, auth_scopes=('configuration:write',)):
+        response = flask_app_client.post(
+            '%s' % CONFIG_PATH,
```
Comment (hwindsor) [inline tests/extensions/edm/utils.py:25]: can just replace with CONFIG_PATH

File: tests/extensions/edm/utils.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+"""
+EDM utils
```
Comment (hwindsor) [inline tests/extensions/edm/utils.py:3]: Are these really edm utils or just configuration utils?

File: tests/modules/sightings/resources/test_custom_fields.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=missing-docstring
+
+from tests.modules.sightings.resources import utils as sighting_utils
+from tests.extensions.edm import utils as edm_utils
+from tests import utils as test_utils
+
+
+CONFIG_PATH = '/api/v1/configuration/default'
```
Comment (hwindsor) [inline tests/modules/sightings/resources/test_custom_fields.py:9]: I don't think this is used

File: tests/modules/sightings/resources/test_custom_fields.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    from app.modules.encounters.models import Encounter
+    from app.modules.assets.models import Asset
+    from app.modules.asset_groups.models import AssetGroup
+    import datetime
+
+    cfd_id = edm_utils.custom_field_create(flask_app_client, admin_user, 'test_cfd')
+    assert cfd_id is not None
+
+    # we should end up with these same counts (which _should be_ all zeros!)
+    orig_ct = test_utils.multi_count(db, (Sighting, Encounter, Asset, AssetGroup))
```
Comment (hwindsor) [inline tests/modules/sightings/resources/test_custom_fields.py:25]: Every usage of this multi_count has the same params and each test that does so has to import all of the models. 
Would a test_utils.all_count(db) be better and then only that function would need to do all of the immports?

File: tests/modules/sightings/resources/test_custom_fields.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        'startTime': timestamp,
+        'context': 'test',
+        'locationId': 'test',
+        'customFields': {
+            cfd_id: cfd_test_value,
+        },
+        'encounters': [{}],
+    }
+    response = sighting_utils.create_sighting(
+        flask_app_client, researcher_1, expected_status_code=200, data_in=data_in
```
Comment (hwindsor) [inline tests/modules/sightings/resources/test_custom_fields.py:40]: could be replaced by flask_app_client, researcher_1, data_in, 200

Comment (naknomum) [reply tests/modules/sightings/resources/test_custom_fields.py:40]: or even drop the `200` too yes?  ok cleaning up / testing.

File: app/modules/sightings/resources.py (hunk 1 above)
Comment (naknomum) [reply app/modules/sightings/resources.py:500]: it is replicating what all the calls in WBIA do.   i believe there are rare cases where you get a 200 but success=false, at least in wbia.  i have just generally been returning out of houston the response from edm, thus me echoing same behavior here.

File: app/modules/sightings/resources.py (hunk 2 above)
Comment (naknomum) [reply app/modules/sightings/resources.py:489]: again, i have just been following the format we have always expected from wbia.   it is a side-effect of edm mimicking wbia.  houston echoing edm.

File: deploy/codex/README.md (hunk 3 above)
Comment (naknomum) [reply deploy/codex/README.md:48]: this was the linting tools doing.... whatever they are doing with some sort of dangly whitespace.

File: app/modules/sightings/resources.py (hunk 1 above)
Comment (hwindsor) [reply app/modules/sightings/resources.py:500]: I am thinking that we should not expose this (I believe) non standard behaviour at the houston level. 
If houston is supposed to be a clean break from the past would it be better to remove this now?

File: tests/extensions/edm/utils.py (hunk 5 above)
Comment (naknomum) [reply tests/extensions/edm/utils.py:30]: not really; just kinda rolled this by hand real quick.

Comment (naknomum) [reply tests/extensions/edm/utils.py:30]: i also kinda imagined this _generic_ configuration POST thing being used elsewhere (despite it only being used once so far) ... so was guessing other cases would be checked here.  guess i could still use this helper function.

File: tests/modules/sightings/resources/test_custom_fields.py (hunk 8 above)
Comment (naknomum) [reply tests/modules/sightings/resources/test_custom_fields.py:9]: curious the linter didnt pick this up.  :thinking:

File: tests/modules/sightings/resources/test_custom_fields.py (hunk 9 above)
Comment (naknomum) [reply tests/modules/sightings/resources/test_custom_fields.py:25]: at this point, yeah.  it seems like (speaking for myself anyway!) i just test all of these classes each time.   might chunk this into the next pr cuz this one is getting kinda stale, heh

**Review by colinwkingen — APPROVED** [review]
LGTM
